src/main/java/com/palantir/ri/ResourceIdentifier.java

```diff
@@ -222,6 +225,7 @@ public static boolean isValidLocator(String locator) {
      * <a href="http://docs.oracle.com/javaee/7/api/javax/ws/rs/PathParam.html">
      * query and path parameters</a>
      */
+    @JsonCreator
```

Comment:
if we're going to include this we can mark all of service, instance, type and locator as final.